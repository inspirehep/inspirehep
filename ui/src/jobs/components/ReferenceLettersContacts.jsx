import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Map } from 'immutable';
import URLList from '../../common/components/URLList';
import EmailList from './EmailList';
import { InlineUL } from '../../common/components/InlineList';

class ReferenceLettersContacts extends Component {
  render() {
    const { referenceLetters } = this.props;
    const urls = referenceLetters.get('urls');
    const emails = referenceLetters.get('emails');
    const shouldRender = Boolean(urls || emails);
    return (
      shouldRender && (
        <div>
          <strong>Letters of Reference should be sent to: </strong>
          <InlineUL wrapperClassName="di">
            {emails && <EmailList emails={emails} />}
            {urls && <URLList urls={urls} />}
          </InlineUL>
        </div>
      )
    );
  }
}

ReferenceLettersContacts.propTypes = {
  referenceLetters: PropTypes.instanceOf(Map),
};
ReferenceLettersContacts.defaultProps = {
  referenceLetters: Map(),
};
export default ReferenceLettersContacts;
