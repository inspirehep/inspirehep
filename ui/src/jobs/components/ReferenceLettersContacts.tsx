import React, { Component } from 'react';
import { Map } from 'immutable';
import URLList from '../../common/components/URLList';
import EmailList from './EmailList';
import { InlineUL } from '../../common/components/InlineList';

type OwnProps = {
    referenceLetters?: $TSFixMe; // TODO: PropTypes.instanceOf(Map)
};

type Props = OwnProps & typeof ReferenceLettersContacts.defaultProps;

class ReferenceLettersContacts extends Component<Props> {

static defaultProps = {
    referenceLetters: Map(),
};

  render() {
    const { referenceLetters } = this.props;
    const urls = referenceLetters.get('urls');
    const emails = referenceLetters.get('emails');
    const shouldRender = Boolean(urls || emails);
    return (
      shouldRender && (
        <div>
          <strong>Letters of Reference should be sent to: </strong>
          {/* @ts-expect-error ts-migrate(2769) FIXME: No overload matches this call. */}
          <InlineUL wrapperClassName="di">
            {emails && <EmailList emails={emails} />}
            {/* @ts-expect-error ts-migrate(2769) FIXME: No overload matches this call. */}
            {urls && <URLList urls={urls} />}
          </InlineUL>
        </div>
      )
    );
  }
}
export default ReferenceLettersContacts;
