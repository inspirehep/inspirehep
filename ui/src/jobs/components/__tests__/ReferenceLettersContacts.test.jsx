import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';
import ReferenceLettersContacts from '../ReferenceLettersContacts';

describe('ReferenceLettersContacts', () => {
  it('renders with referenceLetters', () => {
    const referenceLetters = fromJS({
      urls: [
        { value: 'https://qa.inspirehep.net' },
        { value: 'www.google.com', description: 'Google' },
      ],
      emails: ['awi_moni@yahoo.com', 'mariahmoni@gmail.com'],
    });
    const wrapper = shallow(
      <ReferenceLettersContacts referenceLetters={referenceLetters} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
