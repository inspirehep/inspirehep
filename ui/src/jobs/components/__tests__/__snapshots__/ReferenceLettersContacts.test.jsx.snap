// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReferenceLettersContacts renders with referenceLetters 1`] = `
<div>
  <strong>
    Letters of Reference should be sent to: 
  </strong>
  <InlineUL
    separator=", "
    wrapperClassName="di"
  >
    <EmailList
      emails={
        Immutable.List [
          "awi_moni@yahoo.com",
          "mariahmoni@gmail.com",
        ]
      }
    />
    <URLList
      urls={
        Immutable.List [
          Immutable.Map {
            "value": "https://qa.inspirehep.net",
          },
          Immutable.Map {
            "value": "www.google.com",
            "description": "Google",
          },
        ]
      }
      wrapperClassName={null}
    />
  </InlineUL>
</div>
`;
